### ISSUE TITLE
lflags don't get applied to libs

### ISSUE THREAD

**Issue by vabenil** [MrcSnm/redub#19]
For whatever reason, redub doesn't seem to apply `lflags` for projects with `targetType` `"library"`.

Here I attach a very simple project as an example:
[nuklear_test.zip](https://github.com/user-attachments/files/16324087/nuklear_test.zip)

For security reasons I did not attach the `libmything.a` library file, as it's a binary. To create the `libmythin.a` you can go to `nuklear_test/mything/lib/` and run `make` in there, or just create your own ;) .

I get the following error when compiling. 
```
[ vabenil@arch-pc: /nuklear_test ]$ redub -f -v
Parsing /nuklear_test/dub.json at /nuklear_test with : -c 
Added project nuklear_test to memory cache.
Parsing /nuklear_test/mything/dub.json at /nuklear_test/mything with : -c 
Dependencies resolved in 0 ms for "debug" using /usr/bin/dmd [linux-x86_64]
Project nuklear_test is fully parallelizable! Will build everything at the same time
Built: mything. Took 46ms

        /usr/bin/dmd '-color=on' -op -debug -g '-version=Have_mything' -I/nuklear_test/mything/source /nuklear_test/mything/source/mything.d -lib -od/tmp/.redub -of/nuklear_test/mything/libmything.a 

Built: nuklear_test. Took 51ms

        /usr/bin/dmd '-color=on' -op -debug -g '-version=Have_mything' '-version=Have_nuklear_test' -I/nuklear_test/source -I/nuklear_test/mything/source /nuklear_test/source/app.d -c -od/tmp/.redub -of/nuklear_test/nuklear_test.o 

Linking Error at "nuklear_test". 
        Redub v1.7.9
        dmd 2.109.1
        Failed with flags: 

        /usr/bin/dmd -of/nuklear_test/nuklear_test /nuklear_test/nuklear_test.o -g /nuklear_test/mything/libmything.a
                  :
        /usr/bin/ld: /nuklear_test/nuklear_test.o: in function `_Dmain':
/nuklear_test/source/app.d:5:(.text._Dmain[_Dmain]+0xa): undefined reference to `foo'
collect2: error: ld returned 1 exit status
Error: undefined reference to `foo`
       referenced from `_Dmain`
       perhaps a library needs to be added with the `-L` flag or `pragma(lib, ...)`
Error: linker exited with status 1
       cc /nuklear_test/nuklear_test.o -o /nuklear_test/nuklear_test -g -m64 -Xlinker --export-dynamic /nuklear_test/mything/libmything.a -L/usr/lib -Xlinker -Bstatic -lphobos2 -Xlinker -Bdynamic -lpthread -lm -lrt -ldl 

Warning: If you think this is a bug on redub, do test with dub, if it works, do file an issue at https://github.com/MrcSnm/redub/issues/new/choose
object.Exception@source/redub/api.d(136): Build failure
----------------
source/redub/api.d:136 [0x576eb1005f86]
source/app.d:76 [0x576eb10073c7]
source/app.d:69 [0x576eb1007115]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x76203cfaa06c]
??:? _d_run_main2 [0x76203cfa9e86]
??:? _d_run_main [0x76203cfa9cdc]
??:? [0x76203cccfc87]
??:? __libc_start_main [0x76203cccfd4b]
??:? [0x576eb0fc3e34]
```
If you pay attention, the flag `-lmything` is nowhere.

This compiles normally via dub.

My setup is Arch Linux, and redub is version 1.7.9.

_P.S._ You probably want a test to catch this one

**Comment by MrcSnm** [comment]
Can I see the dub output?

**Comment by vabenil** [comment]
Sure. Paths are different because I removed my home path in the first example for simplicity sake.

dub output:
```
[ vabenil@arch-pc: sync/D/nuklear_test ]$ dub -v
Note: Failed to determine version of package nuklear_test at .. Assuming ~master.
  Version selection for dependency mything (mything) of nuklear_test is missing.
  Adding local mything in /home/vabenil/MEGA/sync/D/nuklear_test/mything
Note: Failed to determine version of package mything at .. Assuming ~master.
  Found dependency mything @./mything
Generating using build
Configuring dependent nuklear_test, deps:"mything"
  Configuring dependent mything, deps:
    Starting Performing "debug" build using /usr/bin/dmd for x86_64.
  Up-to-date mything ~master: target for configuration [library] is up to date.
Using existing build in /home/vabenil/.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ.
Copying target from /home/vabenil/.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ/libmything.a to /home/vabenil/MEGA/sync/D/nuklear_test/mything
File /home/vabenil/MEGA/sync/D/nuklear_test/dub.selections.json doesn't exist, triggering rebuild.
    Building nuklear_test ~master: building configuration [application]
[cwd=/home/vabenil/MEGA/sync/D/nuklear_test] /usr/bin/dmd -debug -g -w -version=Have_nuklear_test -version=Have_mything -Isource/ -Imything/source/ source/app.d -c -of/home/vabenil/.dub
/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test.o -vcolumns
     Linking nuklear_test
[cwd=/home/vabenil/MEGA/sync/D/nuklear_test] /usr/bin/dmd -of/home/vabenil/.dub/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test /home/vabenil/.dub
/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test.o ../../../../.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ/libmything.a 
-L--no-as-needed -L-L/home/vabenil/MEGA/sync/D/nuklear_test/mything/lib -L-lmything -g
Copying target from /home/vabenil/.dub/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test to /home/vabenil/MEGA/sync/D/nuklear_test
    Finished To force a rebuild of up-to-date targets, run again with --force
     Running nuklear_test 
foo 5!
```

**Comment by MrcSnm** [comment]
Oh do test it with `--force` plz

**Comment by vabenil** [comment]
```
[ vabenil@arch-pc: sync/D/nuklear_test ]$ dub -f -v
Note: Failed to determine version of package nuklear_test at .. Assuming ~master.
  Version selection for dependency mything (mything) of nuklear_test is missing.
  Adding local mything in /home/vabenil/MEGA/sync/D/nuklear_test/mything
Note: Failed to determine version of package mything at .. Assuming ~master.
  Found dependency mything @./mything
Generating using build
Configuring dependent nuklear_test, deps:"mything"
  Configuring dependent mything, deps:
    Starting Performing "debug" build using /usr/bin/dmd for x86_64.
    Building mything ~master: building configuration [library]
[cwd=/home/vabenil/MEGA/sync/D/nuklear_test] /usr/bin/dmd -debug -g -w -version=Have_mything -Imything/source/ mything/source/mything.d -L-L/home/vabenil/MEGA/sync/D/nuklear_test/mythin
g/lib -L-lmything -lib -of/home/vabenil/.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ/libmything.a -vcolumns
Copying target from /home/vabenil/.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ/libmything.a to /home/vabenil/MEGA/sync/D/nuklear_test/mything
    Building nuklear_test ~master: building configuration [application]
[cwd=/home/vabenil/MEGA/sync/D/nuklear_test] /usr/bin/dmd -debug -g -w -version=Have_nuklear_test -version=Have_mything -Isource/ -Imything/source/ source/app.d -c -of/home/vabenil/.dub
/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test.o -vcolumns
     Linking nuklear_test
[cwd=/home/vabenil/MEGA/sync/D/nuklear_test] /usr/bin/dmd -of/home/vabenil/.dub/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test /home/vabenil/.dub
/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test.o ../../../../.dub/cache/mything/~master/build/library-debug-mUm__ujiaQ42jYdB6j-FpQ/libmything.a 
-L--no-as-needed -L-L/home/vabenil/MEGA/sync/D/nuklear_test/mything/lib -L-lmything -g
Copying target from /home/vabenil/.dub/cache/nuklear_test/~master/build/application-debug-rhX1mGA1VRt2BfjEX-h9Dw/nuklear_test to /home/vabenil/MEGA/sync/D/nuklear_test
     Running nuklear_test 
foo 5!
```

**Comment by MrcSnm** [comment]
Okay I got the problem now, the lflags doesn't appear to be transferring from leaf to root which causes the problem:

FIY: -L-Lmything and -lmything at the build command is a NO-OP thing, the problem is on the linking line which is not including the LFLAGS

**Comment by MrcSnm** [comment]
The issue might be fixed by v1.7.10
